### ISSUE TITLE
With multiple input files, preventing VimTeX from asking to choose main file each time a different input file is opened

### ISSUE THREAD

**Issue by TryerGit** [lervag/vimtex#2595]
Hello,

Consider the following:

.vimrc:

```vim
if empty(glob('~/.vim/autoload/plug.vim'))
    silent !curl -fLo ~/.vim/autoload/plug.vim --create-dirs
        \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
    autocmd VimEnter * PlugInstall --sync | source $MYVIMRC
endif
call plug#begin('~/.vim/plugged')
Plug 'lervag/vimtex'
call plug#end()

let g:vimtex_quickfix_method = 'pplatex'
```
professor.tex and student.tex (both exactly same):

```tex
\documentclass{article}
\begin{document}
\input{main.tex} 
\end{document}
```
main.tex:

```tex
\input{section1.tex}
\input{section2.tex}
```
section1.tex

```tex
Purposely make an error: miss the closing dollar sign. 

$1 + 4 = 5

$1 != 4$
```
section2.tex

```tex
Open section2.tex directly via vim section2.tex
```

(Perhaps the above has one extra level of indirection? But in any case it was the minimal example I could come up with in this case)

Open section2.tex directly via `vim section2.tex` and choose professor.tex as the main file. Try to compile via `\ll`, `pplatex` correctly identifies the following error in section1.tex:

`./section1.tex|4 error| Missing $ inserted <inserted text>  $  `

When navigating to the quickfix list and pressing enter on this error, one is again presented with an option to choose the main file: professor.tex or student.tex

However, on my machine, neither of these two options are displayed. Please see attached animated gif:

![genarr](https://user-images.githubusercontent.com/70434658/205501693-333cc220-a91c-49db-9e50-fdaccd58f38d.gif)

On pressing enter twice, one is presented with a third "Press Enter or type command to continue" option. However, pressing enter does nothing here. On pressing Ctrl-C, there is an error:

```vim
Error detected while processing /home/TryerGit/.vim/plugged/vimtex/ftplugin/tex.vim:
line   25:
Interrupted
```

(Q1) It appears to me that on opening section2.tex and having chosen the main file as professor.tex, navigating to an error in section1.tex should not ask the user again to choose the main file -- it should automatically use professor.tex as the main file.

(Q2) Even if it is asked, the options don't seem to be displayed on my machine -- please see gif above.

Thank you.

**Comment by lervag** [comment]
> (Q1) It appears to me that on opening section2.tex and having chosen the main file as professor.tex, navigating to an error in section1.tex should not ask the user again to choose the main file -- it should automatically use professor.tex as the main file.

I agree, this was also the intent. Due to the well described issue report, I found a bug and fixed it, and I believe it may have been the cause of the problem here.

> (Q2) Even if it is asked, the options don't seem to be displayed on my machine -- please see gif above.

Yes, in some cases, the menu's are not displayed properly. I believe this is mostly _not_ an issue, e.g. in this case, I think I've fixed the real issue so the second menu does not appear.

However, it is true that the choice menus are not always reliably, and I have not found a very good way to solve that right now. In the meantime, I do think this should more or less not be a real problem.
